src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2266,9 +2266,11 @@ public static function unguarded(callable $callback)
 
         static::unguard();
 
-        $result = $callback();
-
-        static::reguard();
+        try {
+            $result = $callback();
+        } finally {
+            static::reguard();
+        }
 
         return $result;
```

Comment:
I think it does not matter. When the callback throws an exception, this method throws that exception and thus does not return anything. A method either returns or throws an exception. Not both. Or am I wrong now?